### PR TITLE
Add RSA AES key wrap mechanism parameters

### PR DESCRIPTION
**Pull request by varder**
Mechanism implementation:
https://github.com/OP-TEE/optee_os/pull/5647

pkcs11-spec-v3.1-cs01
6.1.23 RSA AES KEY WRAP

The RSA AES key wrap mechanism based on the RSA public-key cryptosystem and the AES key wrap mechanism. It supports single-part key wrapping and key unwrapping.

How to use:
```
params := &pkcs11.RSAAESKeyWrapParams{
	AESKeyBits: 256,
	OAEPParams: pkcs11.OAEPParams{
		HashAlg:    pkcs11.CKM_SHA256,
		MGF:        pkcs11.CKG_MGF1_SHA256,
		SourceType: pkcs11.CKZ_DATA_SPECIFIED,
	},
}
mechanism := []*pkcs11.Mechanism {
    pkcs11.NewMechanism(pkcs11.CKM_RSA_AES_KEY_WRAP, params)
}
unwrappedKey, err := ctx.UnwrapKey(session, mechanism, wrappingKeyObj,
                                   wrappedKey, unwrappedKeyAttributes)
```

### CODE REVIEW

**Comment by varder** [comment]
Hi @miekg,
Could you please have a look at the PR

File: params.go [hunk 1]
```diff
@@ -26,6 +26,11 @@ static inline void putECDH1PublicParams(CK_ECDH1_DERIVE_PARAMS_PTR params, CK_VO
 	params->pPublicData = pPublicData;
 	params->ulPublicDataLen = ulPublicDataLen;
 }
+
+static inline void putRsaAesKeyWrapParams(CK_RSA_AES_KEY_WRAP_PARAMS_PTR params, CK_VOID_PTR pOAEPParams)
```
Comment (miekg) [inline params.go:30]:
```suggestion
static inline void putRSAAESKeyWrapParams(CK_RSA_AES_KEY_WRAP_PARAMS_PTR params, CK_VOID_PTR pOAEPParams)
```
to keep in-line with the usage of RSA and AES in this pkg

File: params.go [hunk 2]
```diff
@@ -188,3 +193,23 @@ func cECDH1DeriveParams(p *ECDH1DeriveParams, arena arena) ([]byte, arena) {
 
 	return memBytes(unsafe.Pointer(&params), unsafe.Sizeof(params)), arena
 }
+
+type RsaAesKeyWrapParams struct {
```
Comment (miekg) [inline params.go:197]: same: s/Rsa/RSA en for AES

File: params.go [hunk 3]
```diff
@@ -188,3 +193,23 @@ func cECDH1DeriveParams(p *ECDH1DeriveParams, arena arena) ([]byte, arena) {
...
+	AESKeyBits uint
+	OAEPParams OAEPParams
+}
+
+func cRsaAesKeyWrapParams(p *RsaAesKeyWrapParams, arena arena) ([]byte, arena) {
```
Comment (miekg) [inline params.go:202]: here too

**Comment by miekg** [comment]
some style stuff, otherwise lgtm

File: params.go (hunk 1 above)
Comment (varder) [reply params.go]: fixed

File: params.go (hunk 2 above)
Comment (varder) [reply params.go]: fixed

File: params.go (hunk 3 above)
Comment (varder) [reply params.go]: fixed

**Pull request merged**